github-runner-manager/src/github_runner_manager/templates/openstack-userdata.sh.j2

```diff
@@ -70,10 +70,10 @@ systemctl restart docker
 # Prepare metrics
 su - ubuntu -c 'mkdir "{{ metrics_exchange_path }}"'
 
-# Insert pre-job script
-cat << 'EOF' | su - ubuntu -c 'tee /home/ubuntu/actions-runner/pre-job.sh'
+# Insert pre-job script, we use a special end marker and not EOF or EOT to avoid that the same is reused inside the script
```

Comment:
What is the special end marker for?
I am not sure what it is for after reading the comment.

Reply:
If the pre-job script contains itself "EOF" the script would not be inserted fully but instead the commands would run.

Therefore I am using a UUID for the delimiter, because it decreases the chance that this delimiter is used itself in the pre-job script.

I also find this a bit hacky and we should consider using cloud-init config data (https://cloudinit.readthedocs.io/en/latest/explanation/format.html#cloud-config-data) instead of user data script, which would allow writing files. But due to the urgency of this PR, I haven't tackled it in here.

Reply:
I'll add a more explanatory comment in a follow-up PR.